sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -387,6 +388,12 @@ private object CosmosSchemaInferenceConfig {
     parseFromStringFunction = include => include.toBoolean,
     helpMessage = "Whether schema inference should include the system properties in the schema")
 
+  private val inferSchemaForceNullableProperties = CosmosConfigEntry[Boolean](key = "spark.cosmos.read.inferSchemaForceNullableProperties",
+    mandatory = false,
+    defaultValue = Some(false),
+    parseFromStringFunction = include => include.toBoolean,
+    helpMessage = "Whether schema inference should enforce inferred properties to be nullable - even when no null-values are contained in the sample set")
```

Comment:
what's the use case for this?

Reply:
Use-case is when a dataset contains a column that in the vast majority of cases is non-null (tripType in the NYC dataset for example). So when using schema inference the sample set might not have a single row with null in the property - in that case the schema inference by default will make the property non-nullable (that was one of the user study feedbacks). This config setting allows to relax that and make properties created by schema inference nullable by default.